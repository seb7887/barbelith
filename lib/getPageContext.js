import { SheetsRegistry } from 'jss';
import createMuiTheme from '@material-ui/core/styles/createMuiTheme';
import createGenerateClassName from '@material-ui/core/styles/createGenerateClassName';

// App color theme
const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
    suppressDeprecationWarnings: true,
    fontFamily: ['"Segoe UI"', 'sans-serif']
  },
  palette: {
    primary: {
      main: '#fff'
    },
    secondary: {
      main: '#000'
    },
    favoriteIcon: {
      main: '#fc766a'
    },
    type: 'light'
  }
});

function createPageContext() {
  return {
    theme,
    // this is needed in order to deduplicate the injection of css in the page
    sheetsManager: new Map(),
    // this is needed in order to inject the critical css
    sheetsRegistry: new SheetsRegistry(),
    // the standard class name generator
    generateClassName: createGenerateClassName()
  };
}

export default function getPageContext() {
  // make sure to create a new context for every server-side request so that data
  // isn't shared between connections (really really BAD)
  if (!process.browser) {
    return createPageContext();
  }

  // reuse context on the client-side
  if (!global.__INIT_MATERIAL_UI__) {
    global.__INIT_MATERIAL_UI__ = createPageContext();
  }

  return global.__INIT_MATERIAL_UI__;
}
