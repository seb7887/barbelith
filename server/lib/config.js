require('dotenv').config();

const dev = process.env.NODE_ENV !== 'production';
const port = process.env.PORT || 3000;

const config = {
  port: process.env.PORT || 3000,
  db: dev ? process.env.DEV_DB : process.env.MONGO_URI,
  url: dev ? `http://localhost:${port}` : process.env.PROD_URL,
  secret: process.env.SESSION_SECRET || 'jwtSecret'
};

module.exports = config;
