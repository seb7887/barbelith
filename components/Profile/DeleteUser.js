import React from 'react';
import IconButton from '@material-ui/core/IconButton';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Delete from '@material-ui/icons/Delete';

import { signoutUser } from '../../lib/auth';
import { deleteUser } from '../../lib/api';

class DeleteUser extends React.Component {
  state = {
    open: false,
    isDeleting: false
  };

  handleDeleteUser = () => {
    const { user } = this.props;

    this.setState({ isDeleting: true });
    deleteUser(user._id)
      .then(() => {
        signoutUser();
      })
      .catch(err => {
        this.setState({ isDeleting: false })
      });
  }

  handleOpen = () => this.setState({ open: true });

  handleClose = () => this.setState({ open: false });

  render() {
    const { open, isDeleting } = this.state;
    return (
      <div>
        {/* Delete Button */}
        <IconButton
          data-testid='delete'
          onClick={this.handleOpen}
          color='secondary'>
          <Delete />
        </IconButton>

        {/* Delete User Dialog */}
        <Dialog data-testid='dialog' open={open} onClose={this.handleClose}>
          <DialogTitle>Delete Account</DialogTitle>
          <DialogContent>
            <DialogContentText>
              Are you sure to delete your account?
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button
              data-testid='button-cancel'
              onClick={this.handleClose}
              color='secondary'>
              Cancel
            </Button>
            <Button
              data-testid='button-confirm'
              onClick={this.handleDeleteUser}
              color='secondary'
              disabled={isDeleting}
            >
              {isDeleting ? 'Deleting' : 'Confirm'}
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default DeleteUser;
