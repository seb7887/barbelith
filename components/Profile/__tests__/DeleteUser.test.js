/* eslint-env jest */
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';

import { render, fireEvent, waitForElement } from 'react-testing-library';
import { deleteUser as mockedDeleteUser } from '../../../lib/api';
import DeleteUser from '../DeleteUser';

jest.mock('../../../lib/api', () => {
  return {
    deleteUser: jest.fn(() => Promise.resolve())
  };
});

afterEach(() => {
  mockedDeleteUser.mockClear();
});

describe('<DeleteUser/>', () => {
  it('renders and matches snapshot', () => {
    const { asFragment } = render(
      <DeleteUser user={{}} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should cancel the confirmation dialog', async () => {
    const { getByTestId } = render(
      <DeleteUser user={{}} />
    );

    // click delete button
    fireEvent.click(getByTestId('delete-button'));

    // wait for dialog
    const dialog = await waitForElement(() => getByTestId('dialog'));
    expect(dialog.textContent).toContain('Delete Account');

    fireEvent.click(getByTestId('button-cancel'));
    expect(getByTestId('delete-button')).toBeTruthy();
  });

  it('should delete user if user confirm action', async () => {
    const { getByTestId } = render(
      <DeleteUser user={{}} />
    );

    // click delete button
    fireEvent.click(getByTestId('delete-button'));

    // wait for dialog
    const dialog = await waitForElement(() => getByTestId('dialog'));
    expect(dialog.textContent).toContain('Delete Account');

    fireEvent.click(getByTestId('button-confirm'));
    expect(getByTestId('button-confirm')).toBeDisabled();

    expect(getByTestId('button-confirm').textContent).toContain('Deleting');
    expect(mockedDeleteUser).toHaveBeenCalledTimes(1);
  });

  it('should thrown an error if api call fails', async () => {
    const { getByTestId } = render(
      <DeleteUser user={{}} />
    );

    mockedDeleteUser.mockRejectedValueOnce('Fail!');

    // click delete button
    fireEvent.click(getByTestId('delete-button'));

    // wait for dialog
    const dialog = await waitForElement(() => getByTestId('dialog'));
    expect(dialog.textContent).toContain('Delete Account');

    fireEvent.click(getByTestId('button-confirm'));

    expect(getByTestId('button-confirm').textContent).toContain('Deleting');
    expect(mockedDeleteUser).toHaveBeenCalledTimes(1);
  });
});