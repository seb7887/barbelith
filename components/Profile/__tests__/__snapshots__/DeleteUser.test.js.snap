// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeleteUser/> renders and matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="MuiButtonBase-root-7 MuiIconButton-root-1 MuiIconButton-colorSecondary-4"
      data-testid="delete-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-6"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-10"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-35"
      />
    </button>
  </div>
</DocumentFragment>
`;
